src/hooks/useFilters.js

```diff
@@ -0,0 +1,30 @@
+import React, { useEffect, useReducer } from 'react';
+import { expandFilter } from '../lib/filter';
+
+const load = filter =>
+  Object.entries(filter).map(([id, value]) => ({
+    id,
+    expanded: expandFilter({ [id]: value }),
+  }));
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'refresh':
+      return load(action.value);
+      break;
+  }
+};
+
+const useFilters = filter => {
+  const [filterForms, dispatch] = useReducer(reducer, []);
```

Comment:
I might be misreading this, but `filterForms` is not a list of `FilterForm` components is it? It's just the list of filter objects AFAICT. I'm guessing this is a leftover from some prior code?

It makes this read a bit strange above:

```
      {filterForms.map((filter, index) => (
        <li key={index}>
          <FilterForm filter={filter} />
        </li>
```

because it's like it's passing a `filterForm` as a `filter` prop to `FilterForm`.

Reply:
good point, I'm not great at coming up with names. This is a "processed" array of filters from the `useFilter` hook. How about `filterList` ?

Reply:
or `filters`